velox/substrait/tests/VeloxSubstraitRoundTripPlanConverterTest.cpp

```diff
@@ -287,6 +286,35 @@ TEST_F(VeloxSubstraitRoundTripPlanConverterTest, ifThen) {
   assertPlanConversion(plan, "SELECT if (c0=1, c0 + 1, c1 + 2) as x FROM tmp");
 }
 
+TEST_F(VeloxSubstraitRoundTripPlanConverterTest, subField) {
+  VectorMaker vectorMaker_{pool_.get()};
+  auto vectors = vectorMaker_.rowVector({
+      vectorMaker_.flatVector<int64_t>({0, 1, 2}),
+      vectorMaker_.flatVector<int8_t>({0, 1, 2}),
+      vectorMaker_.flatVector<int16_t>({0, 1, 1}),
+      vectorMaker_.flatVector<float>({0.1, 1.2, 1.3}),
+      vectorMaker_.flatVector<double>({0.1, 1.2, 1.3}),
+  });
+
+  createDuckDbTable({vectors});
+  auto plan = PlanBuilder()
+                  .values({vectors})
+                  .project({"row_constructor(c1, c2) AS c1_c2"})
+                  .project({"c1_c2.c1", "c1_c2.c2"})
+                  .planNode();
+  assertPlanConversion(plan, "SELECT c1, c2 FROM tmp");
+
+  plan = PlanBuilder()
+             .values({vectors})
+             .project({"row_constructor(c0, c1) AS c0_1", "c2"})
```

Comment:
This is a bit hard to read. Consider,

```
row_constructor(a, b) as ab
row_constructor(ab, c) as abc
abc.ab.a, abc.ab.b, abc.c
```

Reply:
I would like to use this way “abc.ab.a, abc.ab.b, abc.c”. But 
After debugging with gdb, I found these code will automatically change "ab" to "c1".
https://github.com/facebookincubator/velox/blob/main/velox/parse/TypeResolver.cpp#L86-L94.

And I use code like follow to create a nested-row.
![image](https://user-images.githubusercontent.com/14065538/195769780-fa54d42a-fac6-4089-8486-bcff85a38cd8.png)
This way can not work too, because parser can not create table.
error info :
CREATE TABLE tmp(abc ROW)
                                ^, Source: RUNTIME, ErrorCode: INVALID_STATE

So , I add some comments to clarify.

Reply:
@zhejiangxiaomai You can use cast to provide names for struct fields. 

```
  auto plan =
      PlanBuilder()
          .values({data})
          .project({"row_constructor(a, b)::struct(a bigint, b bigint) as ab", "c"})
          .project({"row_constructor(ab, c)::struct(ab struct(a bigint, b bigint), c bigint) as abc"})
          .project({"(abc).ab.a", "(abc).ab.b", "abc.c"})
          .planNode();
```

or

```
  auto plan =
      PlanBuilder()
          .values({data})
          .project(
              {"cast(row_constructor(a, b) as row(a bigint, b bigint)) as ab",
               "c"})
          .project(
              {"cast(row_constructor(ab, c) as row(ab row(a bigint, b bigint), c bigint)) as abc"})
          .project({"(abc).ab.a", "(abc).ab.b", "abc.c"})
          .planNode();
```

Reply:
Thank you for your suggestion. Our server was shut down this weekend. I will try this code at Monday.

Reply:
Hi, @mbasmanova.  I do not directly use this syntax like
 `.project({"(abc).ab.a", "(abc).ab.b", "abc.c"})`, because UT can not pass. The function "toSubstraitExpr"  which I Implemented just support to find exprName from input's child. 
And I use 
`.project({"(abc).ab", "abc.c"})
          .project({"(ab).a", "(ab).b", "c"})’`.
So we have a question whether we need to support nested search？ like find a in abc.  
The structure of inputType abc is ""ROW<abc:ROW<ab:ROW<a:BIGINT,b:BIGINT>,c:BIGINT>>""
And "a is not the child of abc, but the child of ab." 

After my discussion with @rui-mo, we find this comments [here](https://github.com/substrait-io/substrait/blob/main/proto/substrait/algebra.proto#L846-L853) explain the ReferenceSegment
represent "the ordinal position of the desired field". So we deem that it is undesired to support nested search. 

What do you think？If you think it's necessary, I can use recursion to implement it.

Reply:
@zhejiangxiaomai We need to support "(abc).ab.a", but we do no need to support "(abc).a". The former is just a FieldAccessExpr wrapped around another FieldAccessExpr.

Reply:
> @zhejiangxiaomai We need to support "(abc).ab.a", but we do no need to support "(abc).a". The former is just a FieldAccessExpr wrapped around another FieldAccessExpr.

Ok, I got it. I need to do some debugging to confirm why I encounter this situation.  The toSubstraitExpr want me to find "a" from "abc", and it throw "VELOX_USER_FAIL("exprName :{} no name matched!", exprName);"。